Chummer/sheets/Notes set.xslt

```diff
@@ -238,6 +215,191 @@
+            </table>
+          </div>
+          <xsl:call-template name="RowSummary">
+            <xsl:with-param name="text" select="$lang.Armor"/>
+            <xsl:with-param name="blockname" select="'ArmorBlock'"/>
+          </xsl:call-template>
+        </xsl:if>
+
+        <xsl:variable name="rangedlist" select="weapons/weapon[type = 'Ranged' and (notes != '' or accessories/accessory/notes != '' or mods/weaponmod/notes != '')]"/>
+        <xsl:if test="$rangedlist">
+          <div class="block" id="RangedWeaponsBlock">
+            <table><tr><td/></tr></table>
+            <xsl:call-template name="TableTitle">
+              <xsl:with-param name="name" select="$lang.RangedWeapons"/>
+            </xsl:call-template>
+            <table class="tablestyle">
+              <tr>
+                <th width="90%" style="text-align: left">
+                  <xsl:value-of select="$lang.Weapon"/>
+                </th>
+                <th width="10%"/>
+              </tr>
+              <xsl:for-each select="weapons/weapon[type = 'Ranged' and (notes != '' or accessories/accessory/notes != '' or mods/weaponmod/notes != '')]">
+                <xsl:sort select="fullname"/>
+                <xsl:call-template name="weaponnotes"/>
+              </xsl:for-each>
+            </table>
+          </div>
+          <xsl:call-template name="RowSummary">
+            <xsl:with-param name="text" select="$lang.RangedWeapons"/>
+            <xsl:with-param name="blockname" select="'RangedWeaponsBlock'"/>
+          </xsl:call-template>
+        </xsl:if>
+
+        <xsl:variable name="meleelist" select="weapons/weapon[type = 'Melee' and (notes != '' or accessories/accessory/notes != '' or mods/weaponmod/notes != '')]"/>
+        <xsl:if test="$meleelist">
+          <div class="block" id="MeleeWeaponsBlock">
+            <table><tr><td/></tr></table>
+            <xsl:call-template name="TableTitle">
+              <xsl:with-param name="name" select="$lang.MeleeWeapons"/>
+            </xsl:call-template>
+            <table class="tablestyle">
+              <tr>
+                <th width="90%" style="text-align: left">
+                  <xsl:value-of select="$lang.Weapon"/>
+                </th>
+                <th width="10%"/>
+              </tr>
+              <xsl:for-each select="weapons/weapon[type = 'Melee' and (notes != '' or accessories/accessory/notes != '' or mods/weaponmod/notes != '')]">
+                <xsl:sort select="fullname"/>
+                <xsl:call-template name="weaponnotes"/>
+              </xsl:for-each>
+            </table>
+          </div>
+          <xsl:call-template name="RowSummary">
+            <xsl:with-param name="text" select="$lang.MeleeWeapons"/>
+            <xsl:with-param name="blockname" select="'MeleeWeaponsBlock'"/>
+          </xsl:call-template>
+        </xsl:if>
+
+        <xsl:variable name="gearlist" select="gears/gear[iscommlink != 'True' and (notes != '' or children/gear/notes != '')]"/>
```

Comment:
Keep in mind, gear can theoretically nest an infinite number of times, and gear parents (e.g. cyberware) can also sometimes nest an infinite number of times. You want a selection that gets every single descendant, not just the first children.